src/renderers/WebGLRenderer.js

```diff
@@ -1971,7 +2031,20 @@ class WebGLRenderer {
 
 				// common camera uniforms
 
+				if ( capabilities.reverseDepthBuffer ) {
+
+					toReversedProjectionMatrix( camera.projectionMatrix );
+
+				}
+
 				p_uniforms.setValue( _gl, 'projectionMatrix', camera.projectionMatrix );
+
+				if ( capabilities.reverseDepthBuffer ) {
+
+					toForwardProjectionMatrix( camera.projectionMatrix );
```

Comment:
Instead of recomputing the old matrix, couldn't you store the previous projection matrix in a temporary variable and then copy it back? Something like:
```js
camera.projectionMatrix.copy( currentProjectionMatrix );
```

Reply:
Sure, c67c4af631cffc9bf6cf27f941f1f40385afff28. There might be a slightly faster way with https://github.com/mrdoob/three.js/pull/29445#discussion_r1767708597, but this is going to be as good as it's going to get for re-usability.